### PR TITLE
Enable permissive `max-len` and `max-lines` as warning

### PR DESCRIPTION
**Pull request by fregante**
- Closes https://github.com/xojs/xo/issues/738

I think it might make sense to enable them as warning to look at the pushback without creating too much trouble. 

I think 200ch is a good upper limit, but I'm not sure about "1000 lines", especially in files with inline documentation and licensing.

https://eslint.org/docs/v8.x/rules/max-lines
https://eslint.org/docs/v8.x/rules/max-len

### CODE REVIEW

**Comment by sindresorhus** [comment]
I think 2000 lines is too little.

**Comment by fregante** [comment]
The point of this warning is to push people to split up larger files. Even you reach the limit, you can just disable it in the file, hopefully as an exception rather than a common occurrence. 

2000-lines is already excessive to me, but it could be a start to verify any pushback.

**Comment by fregante** [comment]
For the record, this is what a 3000-lines file looks like. In a world of modules, this file would not have any reason to exist.

https://github.com/joshuacline/windick/blob/main/windick.cmd

File: index.js [hunk 1]
```diff
@@ -407,6 +407,14 @@ module.exports = {
 			},
 		],
 		'max-depth': 'warn',
+		'max-len': [
+			'warn',
+			200,
+		],
```
Comment (sindresorhus) [inline index.js:413]: Needs the `ignoreComments: true` option. (unclear to me whether it is the default or not)

File: index.js [hunk 2]
```diff
@@ -407,6 +407,21 @@ module.exports = {
...
+				code: 200,
+				ignoreComments: true,
+				ignoreUrls: true
+			},
+		],
+		'max-lines': [
+			'warn',
+			{
+				max: 2000,
+				skipComments: true,
```
Comment (fregante) [inline index.js:422]: `max-lines` also has a way to exclude comments. I think we can lower the counter to 1500 thanks to this.

```suggestion
		'max-lines': [
			'warn',
			{
				max: 1500,
				skipComments: true,
```

File: index.js (hunk 1 above)
Comment (fregante) [reply index.js:419]: Ignored URLs as well

**Pull request merged**